### ISSUE TITLE
get{Any}Props functions do not play nicely with TypeScript and React Native

### ISSUE THREAD

**Issue by kimthompson** [downshift-js/downshift#1563]
- `downshift` version: v8.2.3
- `node` version: v21.5.0
- `npm` (or `yarn`) version: `yarn` v1.22.21

**Relevant code or config**

```typescript
import Downshift from 'downshift';
import React, {useCallback} from 'react';
import {TextInput} from 'react-native';

export default function Combobox() {
  ...
  return (
    <Downshift itemToString={itemToString}>
      {({
        isOpen,
        getToggleButtonProps,
        getLabelProps,
        getMenuProps,
        getInputProps,
        selectedItem,
        highlightedIndex,
        getItemProps,
        inputValue,
        getRootProps,
      }) => (
          ...
          <TextInput style={styles.input} {...getInputProps({placeholder: 'Choose a title'})} />
          ...
        );
      }}
    </Downshift>
  );
}
```

**What you did**:

I basically pulled [this snack](https://snack.expo.dev/@silviuaavram/downshift-in-react-native) from your docs into my project as a starting point to see if Downshift is the right choice of autocomplete library for us. 

**What happened**:

It works, but our TypeScript compiler is angry, presumably because functions like `getInputProps` return a number of attributes that do not exist on React Native `TextInput`  objects. A few examples of this that I found in the console are:

- aria-activedescendant
- aria-autocomplete
- aria-controls
- aria-labelledby

The exact error is:

> No overload matches this call.
  Overload 1 of 2, '(props: TextInputProps | Readonly<TextInputProps>): TextInput', gave the following error.
    Type '{ onChange?: ChangeEventHandler<Element> | undefined; id: string; 'aria-autocomplete': "list"; 'aria-activedescendant': string | undefined; 'aria-controls': string | undefined; ... 8 more ...; style: { ...; }; }' is not assignable to type 'Readonly<TextInputProps>'.
      Types of property 'onBlur' are incompatible.
        Type 'FocusEventHandler<Element> | undefined' is not assignable to type '((e: NativeSyntheticEvent<TextInputFocusEventData>) => void) | undefined'.
          Type 'FocusEventHandler<Element>' is not assignable to type '(e: NativeSyntheticEvent<TextInputFocusEventData>) => void'.
            Types of parameters 'event' and 'e' are incompatible.
              Property 'relatedTarget' is missing in type 'NativeSyntheticEvent<TextInputFocusEventData>' but required in type 'FocusEvent<Element, Element>'.
  Overload 2 of 2, '(props: TextInputProps, context: any): TextInput', gave the following error.
    Type '{ onChange?: ChangeEventHandler<Element> | undefined; id: string; 'aria-autocomplete': "list"; 'aria-activedescendant': string | undefined; 'aria-controls': string | undefined; ... 8 more ...; style: { ...; }; }' is not assignable to type 'Readonly<TextInputProps>'.
      Types of property 'onBlur' are incompatible.
        Type 'FocusEventHandler<Element> | undefined' is not assignable to type '((e: NativeSyntheticEvent<TextInputFocusEventData>) => void) | undefined'.

**Reproduction repository**:

https://snack.expo.dev/@kimicalburn/downshift-react-native-typings-issue-example

**Problem description**:

tl;dr TypeScript is mad when I try to apply any `get{Any}Props` function to any equivalent React Native component.

**Suggested solution**:

I saw that you proposed that people also submit fixes for their own issues, and I'm game! Though I won't probably really dig into it until after the new year, and even then I'm going to prioritize simply using this library as-is in our app first and foremost. I saw that your current recommendation is to not use the `<Downshift/>` component and to use the hooks instead, so I'm going to attempt to revise that example component accordingly.

I plan to pull down a local version of Downshift and see if I can fix the types, perhaps even providing some alternate return for the functions like `getInputProps` if you're targeting a React Native component. What approach to this would like to see, as the maintainer(s) of this library?

**Comment by silviuaavram** [comment]
Not really a react native user but the types return by those functions
reflect the exact thing that is returned by the functions. In your case,
probably destructuring away what you don’t need from them is the way to go,
and the type issue will be solved as well, right?

**Comment by kimthompson** [comment]
That's another approach I was thinking of! I also just wanted to make sure I wasn't missing anything. If that's the approach you'd rather React Native users take, fair enough.

**Comment by kimthompson** [comment]
^ see above

**Comment by silviuaavram** [comment]
Yeah, I’ve no better idea. Those functions are the same for all platforms.
